cas-server-core-api/src/main/java/org/jasig/cas/ticket/TicketGrantingTicket.java

```diff
@@ -66,11 +66,12 @@
      * @param service The service for which we are granting a ticket
      * @param expirationPolicy the expiration policy.
      * @param credentialsProvided if the credentials are provided.
+     * @param onlyTrackMostRecentSession track the most recent session by keeping the latest service ticket
      * @return the service ticket granted to a specific service for the
      * principal of the TicketGrantingTicket
      */
     ServiceTicket grantServiceTicket(String id, Service service,
-        ExpirationPolicy expirationPolicy, boolean credentialsProvided);
+        ExpirationPolicy expirationPolicy, boolean credentialsProvided, boolean onlyTrackMostRecentSession);
```

Comment:
Yes that's fine by me. 

I have long held the idea that we should have distinct classes for PGT and PT, so that we clean up the API to make that separation known. That effort might help with the problem you encountered here.